internal/client/eth_client.go

```diff
@@ -23,31 +30,84 @@ type EthClientInterface interface {
 	BlockByNumber(ctx context.Context, number *big.Int) (*types.Block, error)
 }
 
-// NewEthClient ... Initializer
-func NewEthClient() EthClientInterface {
+func NewEthClient(restyClient *resty.Client) EthClientInterface {
+	if restyClient == nil {
+		restyClient = resty.New()
+	}
+
+	restyDebug, _ := strconv.ParseBool(os.Getenv("RESTY_DEBUG"))
+	restyClient.SetDebug(restyDebug)
```

Comment:
When moved to the config handling, would be good to assume default as 0 even if the variable doesn't exist.

Reply:
I was a little unsure of how to deal with the config and where it should live; I ended up putting it in the config directory. Please let me know if I did anything wrong and I will make sure to fix it!